contracts/modules/royalty/policies/IpRoyaltyVault.sol

```diff
@@ -280,112 +180,107 @@ contract IpRoyaltyVault is IIpRoyaltyVault, ERC20SnapshotUpgradeable, Reentrancy
+    /// @notice Adds a new revenue token to the vault
     /// @param token The address of the revenue token
-    function pendingVaultAmount(address token) external view returns (uint256) {
-        return _getIpRoyaltyVaultStorage().pendingVaultAmount[token];
-    }
+    function _updateVaultBalance(address token, uint256 amount) internal {
+        IpRoyaltyVaultStorage storage $ = _getIpRoyaltyVaultStorage();
 
-    /// @notice Amount of revenue token in the claim vault
-    /// @param token The address of the revenue token
-    function claimVaultAmount(address token) external view returns (uint256) {
-        return _getIpRoyaltyVaultStorage().claimVaultAmount[token];
-    }
+        if (!ROYALTY_MODULE.isWhitelistedRoyaltyToken(token))
+            revert Errors.IpRoyaltyVault__NotWhitelistedRoyaltyToken();
+        if (amount == 0) revert Errors.IpRoyaltyVault__ZeroAmount();
 
-    /// @notice Amount of revenue token claimable at a given snapshot
-    /// @param snapshotId The snapshot id
-    /// @param token The address of the revenue token
-    function claimableAtSnapshot(uint256 snapshotId, address token) external view returns (uint256) {
-        return _getIpRoyaltyVaultStorage().claimableAtSnapshot[snapshotId][token];
-    }
+        $.tokens.add(token);
+        $.poolInfo[token] += amount;
 
-    /// @notice Indicates whether the claimer has claimed the revenue tokens at a given snapshot
-    /// @param snapshotId The snapshot id
-    /// @param claimer The address of the claimer
-    /// @param token The address of the revenue token
-    function isClaimedAtSnapshot(uint256 snapshotId, address claimer, address token) external view returns (bool) {
-        return _getIpRoyaltyVaultStorage().isClaimedAtSnapshot[snapshotId][claimer][token];
+        emit RevenueTokenAddedToVault(token, amount);
     }
 
-    /// @notice Returns list of revenue tokens in the vault
-    function tokens() external view returns (address[] memory) {
-        return (_getIpRoyaltyVaultStorage().tokens).values();
-    }
+    function _update(address from, address to, uint256 amount) internal override {
+        if (from == address(0)) {
+            super._update(from, to, amount);
+            return;
+        }
+        if (from == to) revert Errors.IpRoyaltyVault__SameFromToAddress(address(this), from);
+        // when transferring RoyaltyTokens (Vault) from a user to another user:
+        // 1. clear pending rewards of the (from) user
+        // 2. clear pending rewards of the (to) another user, if pending rewards of another user is not zero
+        // 3. update the rewardDebt of the (to) another user to accBalancePerShare * userAmount
+        // 4. update the rewardDebt of the (from) user to accBalancePerShare * userAmount
+        // 5. transfer the RoyaltyTokens (Vault) from the (from) user to the (to) another user
+        if (balanceOf(from) == 0) revert Errors.IpRoyaltyVault__ZeroBalance(address(this), from);
```

Comment:
`balanceOf(from)` call reads state and is being used multiple times in this function. We can store it in memory variable to save gas

Reply:
yes, good point. store balance into local var.